src/ray/common/ray_config_def.h

```diff
@@ -154,6 +154,14 @@ RAY_CONFIG(int64_t, max_grpc_message_size, 100 * 1024 * 1024)
 // of retries is non zero.
 RAY_CONFIG(int64_t, grpc_server_retry_timeout_milliseconds, 1000)
 
+// Whether to allow HTTP proxy on GRPC clients. Disable HTTP proxy by default since it
+// disrupts local connections. Note that this config item only controls GrpcClient in
+// `src/ray/rpc/grpc_client.h`. Python GRPC clients are not directly controlled by this.
+// NOTE (kfstorm): DO NOT set this config item via `_system_config`, use
```

Comment:
Can we bypass this limitation by checking this config and explicitly set the env var in the worker pool?

Reply:
The worker pool is actually OK because we already handled workers syncing system config from GCS. The problem is that dashboard (started in `ray.init` or `ray start`) and agent (started by Raylet). One thing we can do is that if grpc HTTP proxy is enabled in `ray.init` or `ray start` via system config, we explicitly call `os.environ["RAY_grpc_http_proxy"] = "true"` before starting any subprocesses.

But there's still one corner case, that is the grpc clients created right in the drivers or `ray start` script before starting any subprocesses. e.g. we create a global state accessor instance (which communicates with GCS via grpc) before starting processes on a worker node:

https://github.com/ray-project/ray/blob/f31e4099eda2aa0087fbc03284ae16fc0afcd9e6/python/ray/node.py#L1102-L1117

In this case, configuring `_system_config` is not possible because we disallow setting `_system_config` on worker nodes. If the env variable `RAY_grpc_http_proxy` is not set, the worker node will try to connect to GCS without an HTTP proxy. That's a dead end.